### PR TITLE
Use `torch.nn.utils.clip_grad_norm_`  and add `clip_grad_by_value` support for TPU

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

See title

- `EPSILON` is removed. PyTorch does not have it as an argument. It should not be necessary as gradients are unscaled before clipping. I'm not sure why `MixedPrecisionPlugin` had `EPSILON=1e-5` instead of `1e-6`. Maybe somebody has more info about this.
- improves docs to clarify link to mixed precision.
- Adds support for `clip_grad_value_` on TPU

Related question on torch-xla: https://github.com/pytorch/xla/issues/2884

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: pytorch_lightning/accelerators/tpu.py [hunk 1]
```diff
@@ -20,15 +20,16 @@
 from pytorch_lightning.plugins.precision import MixedPrecisionPlugin
 from pytorch_lightning.plugins.training_type.single_tpu import SingleTPUPlugin
 from pytorch_lightning.plugins.training_type.tpu_spawn import TPUSpawnPlugin
-from pytorch_lightning.utilities import _XLA_AVAILABLE, GradClipAlgorithmType
+from pytorch_lightning.utilities import _TORCH_GREATER_EQUAL_1_5, _XLA_AVAILABLE
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
 
 if _XLA_AVAILABLE:
     import torch_xla.core.xla_model as xm
-    from torch_xla._patched_functions import clip_grad_norm_
```
Comment (kaushikb11) [inline pytorch_lightning/accelerators/tpu.py:28]: Why not use it directly?

Comment (carmocca) [reply pytorch_lightning/accelerators/tpu.py:28]: Thought it was nicer to rely on their `apply_patches` fn, this way we don't need to override the clip by norm method

**Review by kaushikb11 — APPROVED** [review]
LGTM!

File: docs/source/advanced/training_tricks.rst [hunk 2]
```diff
@@ -26,10 +26,14 @@ The effect is a large effective batch size of size KxN.
...
-If ``gradient_clip_algorithm`` option is set to ``value``, which is ``norm`` by default, this will
-`clip the gradient value <https://pytorch.org/docs/stable/nn.html#torch.nn.utils.clip_grad_value_>`_ for each parameter instead.
+Gradient clipping may be enabled to avoid exploding gradients. By default, this will clip the gradient norm by calling
+:func:`torch.nn.utils.clip_grad_norm_` computed over all model parameters together.
+If the Trainer's ``gradient_clip_algorithm`` is set to ``'value'`` (``'norm'`` by default), this will use instead
+:func:`torch.nn.utils.clip_grad_norm_` for each parameter instead.
+
+.. note::
+    If using mixed precision, The ``gradient_clip_val`` does not need to be changed as the gradients are unscaled
+    before applying the clipping function.
```
Comment (awaelchli) [inline docs/source/advanced/training_tricks.rst:36]:
```suggestion
    If using mixed precision, the ``gradient_clip_val`` does not need to be changed as the gradients are unscaled
    before applying the clipping function.
```

File: tests/models/test_tpu.py [hunk 3]
```diff
@@ -238,8 +238,7 @@ def test_tpu_clip_grad_by_value(tmpdir):
     )
 
     model = BoringModel()
-    with pytest.raises(AssertionError):
-        tpipes.run_model_test(trainer_options, model, on_gpu=False, with_hpc=False)
```
Comment (awaelchli) [inline tests/models/test_tpu.py:242]: that test xD

**Comment by awaelchli** [comment]
There was a PR that did the exact opposite: https://github.com/PyTorchLightning/pytorch-lightning/pull/963

**Comment by carmocca** [comment]
> There was a PR that did the exact opposite: #963

We apply the torch-xla patch for TPU, no need to duplicate the code ourselves

Props for the ancient PR reference though

File: CHANGELOG.md [hunk 4]
```diff
@@ -105,6 +105,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added new `EarlyStopping` parameters `stopping_threshold` and `divergence_threshold` ([#6868](https://github.com/PyTorchLightning/pytorch-lightning/pull/6868))
 
 
+- Added `clip_grad_value` support for TPUs ([#7025](https://github.com/PyTorchLightning/pytorch-lightning/pull/7025))
```
Comment (kaushikb11) [inline CHANGELOG.md:108]:
```suggestion
- Added `clip_grad_by_value` support for TPUs ([#7025](https://github.com/PyTorchLightning/pytorch-lightning/pull/7025))
```

**Review by awaelchli — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

File: CHANGELOG.md [hunk 5]
```diff
@@ -10,6 +10,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 ### Added
 
 
+- Added `clip_grad_by_value` support for TPUs ([#7025](https://github.com/PyTorchLightning/pytorch-lightning/pull/7025))
+
+
 ### Changed
```
Comment (carmocca) [inline CHANGELOG.md:18]:
```suggestion

- Changed `clip_grad_norm` to use `torch.nn.utils.clip_grad_norm_` ([#7025](https://github.com/PyTorchLightning/pytorch-lightning/pull/7025))
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -129,6 +132,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Ensure accelerator is valid if running interactively ([#5970](https://github.com/PyTorchLightning/pytorch-lightning/pull/5970))
 - Disabled batch transfer in DP mode ([#6098](https://github.com/PyTorchLightning/pytorch-lightning/pull/6098))
 
+- Changed `clip_grad_norm` to use `torch.nn.utils.clip_grad_norm_` ([#7025](https://github.com/PyTorchLightning/pytorch-lightning/pull/7025))
+
+
```
Comment (carmocca) [inline CHANGELOG.md:137]:
```suggestion
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -10,9 +10,13 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 ### Added
 
 
+- Added `clip_grad_by_value` support for TPUs ([#7025](https://github.com/PyTorchLightning/pytorch-lightning/pull/7025))
+
+
 ### Changed
 
 
+- Changed `clip_grad_norm` to use `torch.nn.utils.clip_grad_norm_` ([#7025](https://github.com/PyTorchLightning/pytorch-lightning/pull/7025))
```
Comment (carmocca) [inline CHANGELOG.md:19]:
```suggestion
- Changed `clip_grad_norm` to use `torch.nn.utils.clip_grad_norm_` ([#7025](https://github.com/PyTorchLightning/pytorch-lightning/pull/7025))


```
